### PR TITLE
(YONK-120) Fix for the 500 error on progress page

### PR DESCRIPTION
**Pull request by marjev**
@vkaracic These are additional tests for YONK-8.

You can check the assertions and also use this branch to add more tests.

### CODE REVIEW

File: lms/djangoapps/gradebook/tests.py [hunk 1]
```diff
@@ -34,6 +36,8 @@
 class GradebookTests(ModuleStoreTestCase):
     """ Test suite for Student Gradebook """
 
+    maxDiff = None
```
Comment (marjev) [inline lms/djangoapps/gradebook/tests.py]: Since this is WIP, I've left this for now. I'll remove this line before merging into master.

**Comment by marjev** [comment]
@smagoun 

Here are some additional tests you've requested. I've noticed a lack of assertions in are of unit testing the receiver of the `on_score_changed` signal, so I've added some to make sure that database is updated appropriately.

Cases that are tested:

1) Score is changed multiple times in a short period of time
2) Score is changed in an open course that has started
3) Score is changed in a course that is not yet started
4) Score is changed in a course that is closed and user is a student/admin/staff
5) Associated gradebook entries are deleted when the course is deleted

**Comment by vkaracic** [comment]
:+1:

**Comment by marjev** [comment]
@vkaracic Please review the code.

JIRA ticket: https://openedx.atlassian.net/browse/YONK-120

Steps to reproduce:
1) Delete a few entries from `edxapp.gradebook_studentgradebook` table
2) Assign `progress summary`, `grade summary` and `grading policy` a blank value for a few entries
3) Run the command `python manage.py lms generate_gradebook_entries --settings="devstack"` to generate gradebook entries
4) Deleted entries should be added back to the `edxapp.gradebook_studentgradebook` table
5) Entries that have blank values should no have values for `progress summary`, `grade summary` and `grading policy`

**Comment by marjev** [comment]
@smagoun This should fix the YONK-120 issue. I tagged Vedran to take a look at it, but would also like to have someone else from a team to give a 2nd review. If you have someone in mind, please tag them.

I reproduced the issue locally and fix it. Wrote the tests for that case too.

**Comment by smagoun** [comment]
@marjev, I suggest having @mattdrayer or @ziafazal review this

**Comment by mattdrayer** [comment]
:+1: on this change, and thanks for all of the additional tests!

**Pull request merged**